File: frontend/src/api/history.ts
/**
 * TextMirror 校对历史 API
 */
import request from '@/utils/request'
import type { ReviewIssue } from '@/utils/review'

export interface HistoryReviewMetadata {
  mode: 'single' | 'compare' | 'collaboration' | null
  coverage_status: 'complete' | 'partial' | 'unknown'
  review_summary: {
    total: number
    accepted: number
    ignored: number
    pending: number
    failed_models: number
  }
}

export interface HistoryItem extends HistoryReviewMetadata {
  id: number
  type: string
  domain: string
  total_issues: number
  text_preview: string
  source_filename?: string
  token_usage?: Record<string, number>
  created_at?: string
}

export interface HistoryListResponse {
  items: HistoryItem[]
  total: number
  page: number
  page_size: number
}

export interface HistoryDetail extends HistoryReviewMetadata {
  issues: ReviewIssue[]
  id: number
  type: string
  domain: string
  original_text: string
  modified_text?: string
  check_types?: string
  result?: {
    issues?: Array<{
      original: string
      type: string
      suggestion: string
      explanation: string
      severity: string
    }>
    versions?: Array<{ label: string; content: string }>
    [key: string]: unknown
  }
  total_issues: number
  source_filename?: string
  token_usage?: Record<string, number>
  created_at?: string
}

/** 今日使用量与配额 */
export interface TodayUsage {
  used_today: number
  daily_quota: number | null  // null = 不限
}

/** 获取今日使用量与配额 */
export function getTodayUsageApi(): Promise<TodayUsage> {
  return request.get('/history/usage')
}

/** 获取历史列表 */
export function listHistoryApi(params?: {
  page?: number
  page_size?: number
  type?: string
  domain?: string
}): Promise<HistoryListResponse> {
  return request.get('/history', { params })
}

/** 获取历史详情 */
export function getHistoryDetailApi(id: number): Promise<HistoryDetail> {
  return request.get(`/history/${id}`)
}

/** 删除历史记录 */
export function deleteHistoryApi(id: number): Promise<void> {
  return request.delete(`/history/${id}`)
}

File: backend/app/schemas/history.py
"""
TextMirror 校对历史 Schema
"""
from datetime import datetime
from typing import Any, Dict, List, Literal, Optional

from pydantic import BaseModel, Field

from app.schemas.review import ReviewIssue


class HistoryReviewSummary(BaseModel):
    """已保存草稿（未保存时为原始报告）的去重问题数，不是补丁数。"""
    total: int = 0
    accepted: int = 0
    ignored: int = 0
    pending: int = 0
    failed_models: int = 0


class HistoryReviewMetadata(BaseModel):
    mode: Literal["single", "compare", "collaboration"] | None = None
    coverage_status: Literal["complete", "partial", "unknown"] = "unknown"
    review_summary: HistoryReviewSummary = Field(default_factory=HistoryReviewSummary)


class HistoryListItem(HistoryReviewMetadata):
    """历史列表项"""
    id: int
    type: str
    domain: str
    total_issues: int
    text_preview: str
    source_filename: Optional[str] = None
    token_usage: Optional[Dict[str, Any]] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class HistoryListResponse(BaseModel):
    """历史列表响应（分页）"""
    items: List[HistoryListItem]
    total: int
    page: int
    page_size: int


class HistoryDetailResponse(HistoryReviewMetadata):
    """历史详情响应；issues 使用与继续审阅相同的合并及用户决策口径。"""
    issues: list[ReviewIssue] = Field(default_factory=list)
    id: int
    type: str
    domain: str
    original_text: str
    modified_text: Optional[str] = None
    check_types: Optional[str] = None
    result: Optional[Dict[str, Any]] = None
    total_issues: int
    source_filename: Optional[str] = None
    token_usage: Optional[Dict[str, Any]] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}
